[SPARK-14184][SQL] Support native execution of SHOW DATABASE command and fix SHOW TABLE to use table identifier pattern

dilipbiswal opened the pull request:
## What changes were proposed in this pull request?

This PR addresses the following
1. Supports native execution of SHOW DATABASES command
2. Fixes SHOW TABLES to apply the identifier_with_wildcards pattern if supplied.

SHOW TABLE syntax

```
SHOW TABLES [IN database_name] ['identifier_with_wildcards'];
```

SHOW DATABASES syntax

```
SHOW (DATABASES|SCHEMAS) [LIKE 'identifier_with_wildcards'];
```
## How was this patch tested?

Tests added in SQLQuerySuite (both hive and sql contexts) and DDLCommandSuite

Note: Since the table name pattern was not working , tests are added in both SQLQuerySuite to
verify the application of the table pattern.

dilipbiswal committed:
[SPARK-14184] Support native execution of SHOW DATABASE command and fix SHOW TABLE to use table identifier pattern

dilipbiswal commented:
cc @gatorsmile @yhuai

AmplabJenkins commented:
Can one of the admins verify this patch?

gatorsmile commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala:
```diff
@@ -350,6 +358,40 @@ case class ShowTablesCommand(databaseName: Option[String]) extends RunnableComma
+    val catalog = sqlContext.sessionState.catalog
+    val databases =
+      if (databasePattern.isDefined) {
+      catalog.listDatabases(databasePattern.get)
```
indents.

gatorsmile commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala:
```diff
@@ -350,6 +358,40 @@ case class ShowTablesCommand(databaseName: Option[String]) extends RunnableComma
+ */
+case class ShowDatabasesCommand(databasePattern: Option[String]) extends RunnableCommand {
+
+  // The result of SHOW TABLES has two columns, tableName and isTemporary.
```
`SHOW TABLES` -> `SHOW DATABASES`

gatorsmile commented:
You also need to add test cases in `SQLContext` for verifying correctness of the Command execution. So far, it only covers the Parsing correctness.

dilipbiswal committed:
Review comments

dilipbiswal commented:
@gatorsmile Thanks for your review. I have addressed most of your comments except the last one. Currently create/drop database commands are not supported for SQLContext. Thus i am unable to do a meaningful test.

gatorsmile commented:
@dilipbiswal Sorry for that. Will try to finish that part ASAP. Thanks!

dilipbiswal committed:
fix1

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala:
```diff
@@ -369,13 +369,33 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
+        // Parses the following conditions
+        // 1. SHOW TABLES
+        // 2. SHOW TABLES FROM database 'table_pattern'
+        // 3. SHOW TABLES FROM database
```
why accept `FROM`? It seems kind of weird. Hive doesn't do this either: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ShowTables

dilipbiswal replied:
@andrewor14 Hello, we always transform to TOK_FROM for both TOK_IN and TOK_FROM in the grammer file.
https://github.com/apache/spark/blob/637a78f1d3dff00658324de3887d75c5ccd857be/sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlParser.g#L1473 
So now, both form works.
```SQL
show tables from default 
show tables in default 
```

I will fix up the comments to use 'IN' as opposed to 'FROM'. Did you want to disallow this syntax in the grammer file ? Would there be a backward-compatibility issue if we do that ?

andrewor14 replied:
I see, that's too bad. Let's just keep it

andrewor14 commented:
ok to test

andrewor14 commented:
FYI this blocks on #12009.

SparkQA commented:
**[Test build #54373 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54373/consoleFull)** for PR 11991 at commit [`1e15bc2`](https://github.com/apache/spark/commit/1e15bc216db893c680565246c31fe68e9ce215fd).

SparkQA commented:
**[Test build #54373 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54373/consoleFull)** for PR 11991 at commit [`1e15bc2`](https://github.com/apache/spark/commit/1e15bc216db893c680565246c31fe68e9ce215fd).
 * This patch **fails to build**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54373/
Test FAILed.

dilipbiswal committed:
thriftserver test fix

dilipbiswal committed:
rebase with antlr4 change

SparkQA commented:
**[Test build #54380 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54380/consoleFull)** for PR 11991 at commit [`92fe39f`](https://github.com/apache/spark/commit/92fe39f2e9f7363c7b7bd915c494ffca2350b52f).

hvanhovell commented:
@dilipbiswal Could you also add support for the ANTLR4 parser? You'll need to a rule to SqlBase.q4 and do the parsing in SparkSqlParser.scala. Let me know if you need help.

dilipbiswal commented:
@hvanhovell Sure Herman. I will give it a try and get back to you with questions :-)

viirya commented:
@hvanhovell Can you please add @dilipbiswal into whitelist? Thanks!

andrewor14 commented:
add to whitelist

SparkQA commented:
**[Test build #54395 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54395/consoleFull)** for PR 11991 at commit [`92fe39f`](https://github.com/apache/spark/commit/92fe39f2e9f7363c7b7bd915c494ffca2350b52f).

SparkQA commented:
**[Test build #54380 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54380/consoleFull)** for PR 11991 at commit [`92fe39f`](https://github.com/apache/spark/commit/92fe39f2e9f7363c7b7bd915c494ffca2350b52f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54380/
Test FAILed.

SparkQA commented:
**[Test build #54395 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54395/consoleFull)** for PR 11991 at commit [`92fe39f`](https://github.com/apache/spark/commit/92fe39f2e9f7363c7b7bd915c494ffca2350b52f).
 * This patch **fails Spark unit tests**.
 * This patch **does not merge cleanly**.
 * This patch adds no public classes.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54395/
Test FAILed.

rxin commented:
I just merged https://github.com/apache/spark/pull/12015

Can you update to use the new ANTLR4 parser instead? We are going to remove the ANTLR3 one in the next day or two.

dilipbiswal committed:
antlr4 based parsing.

dilipbiswal commented:
@rxin Yes.. I am working on using the new ANTLR4 parser. Thank you.

SparkQA commented:
**[Test build #54443 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54443/consoleFull)** for PR 11991 at commit [`b66f16e`](https://github.com/apache/spark/commit/b66f16e25d1208150e0f0fd507d98b51328cd485).

SparkQA commented:
**[Test build #54443 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54443/consoleFull)** for PR 11991 at commit [`b66f16e`](https://github.com/apache/spark/commit/b66f16e25d1208150e0f0fd507d98b51328cd485).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54443/
Test PASSed.

hvanhovell commented on sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/ng/SqlBase.g4:
```diff
@@ -107,7 +107,8 @@ statement
     | DROP TEMPORARY? FUNCTION (IF EXISTS)? qualifiedName              #dropFunction
     | EXPLAIN explainOption* statement                                 #explain
     | SHOW TABLES ((FROM | IN) db=identifier)?
```
You can remove the 'from' option now.

dilipbiswal replied:
Sure. Will do

dilipbiswal replied:
@hvanhovell Actually Herman, i tried the following in Hive. 
```SQL
hive> show tables from default;
OK
default__t1_t1_index__
```
should we keep the from option ? There are also some tests which use from option.

hvanhovell replied:
Lets keep it then.

hvanhovell commented on sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/ng/SqlBase.g4:
```diff
@@ -107,7 +107,8 @@ statement
     | EXPLAIN explainOption* statement                                 #explain
     | SHOW TABLES ((FROM | IN) db=identifier)?
-        (LIKE (qualifiedName | pattern=STRING))?                       #showTables
+        (LIKE? (qualifiedName | pattern=STRING))?                       #showTables
```
Nit `#showTables` is not alligned anymore.

dilipbiswal replied:
Will fix it.

hvanhovell commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala:
```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
-    if (ctx.LIKE != null) {
-      logWarning("SHOW TABLES LIKE option is ignored.")
+    if (ctx.qualifiedName() != null) {
+      logWarning("SHOW TABLES qualifiedName is ignored.")
```
I am not sure any one (besides us) will understand this.

rxin replied:
yea end users would have no idea what qualified names mean

dilipbiswal replied:
@hvanhovell @rxin 
qualifiedName does not seem to work in hive. Only the table name pattern works. Not sure if its a defect in hive...
Just for reference i looked at cloudera documentation on how they support SHOW commands. They just take
a pattern string literal. Should we simplify our grammer and drop the qualifiedName ? Please let me know.
http://www.cloudera.com/documentation/archive/impala/2-x/2-1-x/topics/impala_show.html

hvanhovell replied:
The only reason for qualifiedName to be used was Hive. If Hive doesn't support it  (it is supported in the parser BTW), then we should drop it.

dilipbiswal replied:
@hvanhovell Yeah.. its supported in their parser. But for every input i see a empty result set returned. Need to check if we break some test-case which expects an empty result set on a input qualified name. Let me run the test suites now :-)

hvanhovell commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala:
```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
-    ShowTablesCommand(Option(ctx.db).map(_.getText))
+    ShowTablesCommand(
+      Option(ctx.db).map(_.getText),
+      Option(ctx.pattern).map(p => unquoteString(p.getText)))
```
`Option(ctx.pattern).map(string)` also see my comments in your branch.

dilipbiswal replied:
@hvanhovell Yeah.. i had seen your comment. I was running test with exact same change :-)

hvanhovell commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala:
```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
+   */
+  override def visitShowDatabases(ctx: ShowDatabasesContext): LogicalPlan = withOrigin(ctx) {
+    if (ctx.qualifiedName() != null) {
+      logWarning("SHOW DATABASES qualifiedName is ignored.")
```
I am not sure any one (besides us) will understand this. Any idea how hive treats these?

hvanhovell commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala:
```diff
@@ -350,6 +354,34 @@ case class ShowTablesCommand(databaseName: Option[String]) extends RunnableComma
+    val schema = StructType(
+      StructField("result", StringType, false) :: Nil)
+
+    schema.toAttributes
```
Why not just create the AttributeReference?

dilipbiswal replied:
OK.

dilipbiswal commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala:
```diff
@@ -350,6 +354,34 @@ case class ShowTablesCommand(databaseName: Option[String]) extends RunnableComma
+    val catalog = sqlContext.sessionState.catalog
+    val databases =
+      databasePattern.map(catalog.listDatabases(_)).getOrElse(catalog.listDatabases())
+    val rows = databases.map{d => Row(d)}
```
Will make the change.

dilipbiswal committed:
Herman's comments

SparkQA commented:
**[Test build #54475 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54475/consoleFull)** for PR 11991 at commit [`3e6fb2e`](https://github.com/apache/spark/commit/3e6fb2e30712c423339170d5c6fbeeb9c19bdec8).

gatorsmile commented:
@dilipbiswal The PR https://github.com/apache/spark/pull/12009 has been resolved. Could you add a test case for show database? Thanks!

SparkQA commented:
**[Test build #54475 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54475/consoleFull)** for PR 11991 at commit [`3e6fb2e`](https://github.com/apache/spark/commit/3e6fb2e30712c423339170d5c6fbeeb9c19bdec8).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54475/
Test PASSed.

viirya commented on sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ng/ParserUtils.scala:
```diff
@@ -24,6 +24,7 @@ import org.apache.spark.sql.catalyst.parser.ParseUtils.unescapeSQLString
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.catalyst.trees.{CurrentOrigin, Origin}
 
+
```
?

dilipbiswal replied:
Fixed. Thanks !!

dilipbiswal committed:
Simon and Xiao's comment

SparkQA commented:
**[Test build #54500 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54500/consoleFull)** for PR 11991 at commit [`034af13`](https://github.com/apache/spark/commit/034af13270b4999a8f2b5dc064b0c268fd8db802).

SparkQA commented:
**[Test build #54500 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54500/consoleFull)** for PR 11991 at commit [`034af13`](https://github.com/apache/spark/commit/034af13270b4999a8f2b5dc064b0c268fd8db802).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54500/
Test PASSed.

gatorsmile commented on sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:
```diff
@@ -2376,4 +2376,47 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     }
   }
+
+  test("show tables") {
```
Can you move this test case into `DDLSuite.scala`? `SQLQuerySuite` is already very big now.

dilipbiswal replied:
OK

gatorsmile commented on sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala:
```diff
@@ -31,9 +31,11 @@ class DDLSuite extends QueryTest with SharedSQLContext {
       dbNames.foreach { name =>
         sqlContext.sql(s"DROP DATABASE IF EXISTS $name CASCADE")
       }
+      sqlContext.sessionState.catalog.setCurrentDatabase(currentDb)
```
The value of `currentDb` is?

dilipbiswal replied:
default

gatorsmile replied:
How about doing it directly? `setCurrentDatabase("default")`

Then, you do not need to first get the value for `currentDb` since `currentDb` could be any database name here.

dilipbiswal replied:
@gatorsmile Yeah.,initially i was thinking of resetting to "default". I thought a restore semantics might be better here. Is it possible for us to do nesting in withDatabase ? Like
WithDatabase("foo") {
   // create db
   // sql
   // withDatabase("bar") {
        sql
      }
      sql
      sql
}
Let me know if you want to reset it to "default" instead of restoring ?

gatorsmile replied:
IMO, we do not need nested cases now. If multiple databases are used, we can do something like, 
```Scala
withDatabase("db1", "db2")
```

dilipbiswal replied:
@gatorsmile Ok.. I will make the change.

dilipbiswal committed:
Move show tables to DDLSuiite

SparkQA commented:
**[Test build #54527 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54527/consoleFull)** for PR 11991 at commit [`fe60df8`](https://github.com/apache/spark/commit/fe60df84f7f5bbce9a485381531043ef1fe8db33).

SparkQA commented:
**[Test build #54527 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54527/consoleFull)** for PR 11991 at commit [`fe60df8`](https://github.com/apache/spark/commit/fe60df84f7f5bbce9a485381531043ef1fe8db33).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54527/
Test PASSed.

dilipbiswal committed:
reset database to default

SparkQA commented:
**[Test build #54539 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54539/consoleFull)** for PR 11991 at commit [`9a11daa`](https://github.com/apache/spark/commit/9a11daae879ab595505d6869739020c104141254).

hvanhovell commented on sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/ng/SqlBase.g4:
```diff
@@ -838,6 +839,8 @@ OUTPUTFORMAT: 'OUTPUTFORMAT';
 INPUTDRIVER: 'INPUTDRIVER';
 OUTPUTDRIVER: 'OUTPUTDRIVER';
 DATABASE: 'DATABASE' | 'SCHEMA';
+DATABASES: 'DATABASES';
```
Do we need to seperate tokens here? I thought they were synonyms?

dilipbiswal replied:
@hvanhovell Herman, yeah.. they are synonyms. I was not sure, If we just define it as one Token as say "DATABASES" , we can still refer to it from the nonReserved rule as 
"DATABASES" ?

hvanhovell replied:
Yeah we can. It would be the same as the `DATABASE` lexer rule.

SparkQA commented:
**[Test build #54539 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54539/consoleFull)** for PR 11991 at commit [`9a11daa`](https://github.com/apache/spark/commit/9a11daae879ab595505d6869739020c104141254).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54539/
Test PASSed.

dilipbiswal committed:
one token for DATABASES and SCHEMAS

SparkQA commented:
**[Test build #54554 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54554/consoleFull)** for PR 11991 at commit [`9a4f356`](https://github.com/apache/spark/commit/9a4f3564c3a44db237d05f5fe26d2d2aa043cb69).

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala:
```diff
@@ -370,13 +370,33 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
       case Token("TOK_CLEARCACHE", Nil) =>
         ClearCacheCommand
 
+      // SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards']
```
FYI to avoid conflicts with #12071 let's revert all the changes in this file since we're not using this parser anymore anyway.

dilipbiswal replied:
ok. Sure.

dilipbiswal replied:
@andrewor14 Hi Andrew, is it ok if i have a very minor change in this file just to make sure compilation goes through. Or it will still cause issue with merging ?

ShowTablesCommand(databaseName, None) => we introduced a new parameter to the command.

andrewor14 replied:
I see... never mind then. It seems that we can't avoid the conflicts after all.

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala:
```diff
@@ -70,12 +70,26 @@ class SparkSqlAstBuilder extends AstBuilder {
    * Create a [[ShowTablesCommand]] logical plan.
+   * Example SQL :
+   * {{{
+   *   SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards'];
```
can you add an optional `LIKE` here

dilipbiswal replied:
Thank you @andrewor14 . Will add it now.

andrewor14 commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala:
```diff
@@ -322,18 +322,17 @@ case class DescribeCommand(
  * The syntax of using this command in SQL is:
  * {{{
- *    SHOW TABLES [IN databaseName]
+ *   SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards'];
```
same here

dilipbiswal replied:
Thank you @andrewor14 . Will add it now.

andrewor14 commented:
Looks great!

dilipbiswal committed:
Reset SparkQl and address more comments

SparkQA commented:
**[Test build #54554 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54554/consoleFull)** for PR 11991 at commit [`9a4f356`](https://github.com/apache/spark/commit/9a4f3564c3a44db237d05f5fe26d2d2aa043cb69).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54554/
Test PASSed.

SparkQA commented:
**[Test build #54565 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54565/consoleFull)** for PR 11991 at commit [`2e16ce4`](https://github.com/apache/spark/commit/2e16ce44ff11bcdc588383cdaaaa00175482ccce).

SparkQA commented:
**[Test build #54565 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54565/consoleFull)** for PR 11991 at commit [`2e16ce4`](https://github.com/apache/spark/commit/2e16ce44ff11bcdc588383cdaaaa00175482ccce).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54565/
Test PASSed.

andrewor14 commented:
@dilipbiswal can you rebase? I think `SparkQl` is now deleted.

dilipbiswal commented:
@andrewor14 Hi Andrew, just rebased. - FYI

SparkQA commented:
**[Test build #54651 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54651/consoleFull)** for PR 11991 at commit [`bafca92`](https://github.com/apache/spark/commit/bafca92906aaaa0a62d255f8067e11531d4609da).

SparkQA commented:
**[Test build #54651 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54651/consoleFull)** for PR 11991 at commit [`bafca92`](https://github.com/apache/spark/commit/bafca92906aaaa0a62d255f8067e11531d4609da).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54651/
Test PASSed.

dilipbiswal commented:
cc @andrewor14

hvanhovell commented:
LGTM

hvanhovell commented:
Merging to master. Thanks!

dilipbiswal commented:
@hvanhovell Thank you, Herman.
